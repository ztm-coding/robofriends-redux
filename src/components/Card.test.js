import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Card from "./Card";

test("Card Component matches the snapshot", () => {
    const { asFragment } = render(
        <Card name="John Doe" email="john@example.com" id="123" />
    );

    // Check if name, email, and image are rendered correctly
    expect(screen.getByText("John Doe")).toBeInTheDocument();
    expect(screen.getByText("john@example.com")).toBeInTheDocument();
    expect(screen.getByAltText("robots")).toHaveAttribute(
        "src",
        "https://robohash.org/123?size=200x200"
    );

    // Snapshot test
    expect(asFragment()).toMatchSnapshot(); // Use asFragment() for snapshot testing
});

// Test for handling missing props (id, name, email)
test("Card component handles missing props gracefully", () => {
    const { asFragment } = render(<Card />); // No props passed

    // Test that the name and email are not rendered if they don't exist
    expect(screen.queryByText(/name/i)).toBeNull();
    expect(screen.queryByText(/email/i)).toBeNull();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
});

// Test if the fallback image is used when id is missing
test("Card component renders fallback image when no id is passed", () => {
    render(<Card name="Jane Doe" email="jane@example.com" />);

    // Test that the image src includes "undefined" for the missing id
    expect(screen.getByAltText("robots")).toHaveAttribute(
        "src",
        "https://robohash.org/undefined?size=200x200"
    );
});
