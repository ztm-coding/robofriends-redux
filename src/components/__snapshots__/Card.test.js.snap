// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
    data-testid="card-component"
  >
    <img
      alt="robots"
      height="200"
      src="https://robohash.org/123?size=200x200"
      width="200"
    />
    <div>
      <h2>
        John Doe
      </h2>
      <p>
        john@example.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component handles missing props gracefully 1`] = `
<DocumentFragment>
  <div
    class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
    data-testid="card-component"
  >
    <img
      alt="robots"
      height="200"
      src="https://robohash.org/undefined?size=200x200"
      width="200"
    />
    <div>
      <h2 />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
